src/System.Diagnostics.Process/tests/RemotelyInvokable.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Diagnostics.Tests
+{
+    /// <summary>
+    /// This class is used as interop between uap and netfx app running in the same app container.
+    /// On uap code running here will be running on netfx.
+    /// This is a workaround for a limitation of uap RemoteInvoke which does not give us much control over the process.
+    /// </summary>
+    internal static class RemotelyInvokable
+    {
+        public static readonly int SuccessExitCode = 42;
+        public const int WaitInMS = 30 * 1000;
+        public const string TestConsoleApp = "System.Diagnostics.Process.Tests";
+
+        public static string DummyUapCmd()
+        {
+            return $@"exit {SuccessExitCode}";
+        }
+
+        public static int Dummy()
+        {
+            return SuccessExitCode;
+        }
+
+        public static int LongWait()
+        {
+            Thread.Sleep(WaitInMS);
+            return SuccessExitCode;
+        }
+
+        public static string ExitWithCodeUapCmd(string exitCodeStr)
+        {
+            return $@"exit {int.Parse(exitCodeStr)}";
+        }
+
+        public static int ExitWithCode(string exitCodeStr)
+        {
+            return int.Parse(exitCodeStr);
+        }
+
+        public static string ErrorProcessBodyUapCmd()
+        {
+            return $"(echo {TestConsoleApp} started error stream) 1>&2 & " +
+                $"(echo {TestConsoleApp} closed error stream) 1>&2 & " +
+                $"exit {SuccessExitCode}";
+        }
+
+        public static int ErrorProcessBody()
+        {
+            Console.Error.WriteLine(TestConsoleApp + " started error stream");
+            Console.Error.WriteLine(TestConsoleApp + " closed error stream");
+            return SuccessExitCode;
+        }
+
+        public static string StreamBodyUapCmd()
+        {
+            return $"(echo {TestConsoleApp} started) & " +
+                $"(echo {TestConsoleApp} closed) & " +
+                $"exit {SuccessExitCode}";
+        }
+
+        public static int StreamBody()
+        {
+            Console.WriteLine(TestConsoleApp + " started");
+            Console.WriteLine(TestConsoleApp + " closed");
+            return SuccessExitCode;
+        }
+
+        public static string ReadLineUapCmd()
+        {
+            return "findstr -src:^..*$";
```

Comment:
@tarekgh any propositions? I'm pretty sure it will exist as long as cmd.exe exists since this is most basic tool - should we fix it when we hit this?